### PR TITLE
API: raise on unsupported dtype instead of silently swapping

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Looks like ATM the DataFrame case raises bc it goes through a different path, while the Series case does not.

mentioned in (but this does not close) #48979

### CODE REVIEW

File: doc/source/whatsnew/v2.0.0.rst [hunk 1]
```diff
@@ -99,6 +99,48 @@ notable_bug_fix2
...
+   0   2016-01-01
+   dtype: datetime64[ns]
+
+In pandas 2.0 we support resolutions "s", "ms", "us", and "ns". When passing
+a supported dtype (e.g. "datetime64[s]"), the result now has exactly
+the requested dtype:
+
+*New behavior*:
+
+   In [5]: pd.Series(["2016-01-01"], dtype="datetime64[s]")
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:131]: I think this should be in a `.. ipython:: python` block

File: doc/source/whatsnew/v2.0.0.rst [hunk 2]
```diff
@@ -99,6 +99,52 @@ notable_bug_fix2
...
+   0   2016-01-01
+   dtype: datetime64[s]
+
+With an un-supported dtype, pandas now raises instead of silently swapping in
+a supported dtype:
+
+*New behavior*:
+
+.. ipython:: python
+
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:144]:
```suggestion
    :okexcept:
```

File: doc/source/whatsnew/v2.0.0.rst [hunk 3]
```diff
@@ -99,6 +99,52 @@ notable_bug_fix2
...
+   In [6]: pd.Series(["2016-01-01"], dtype="datetime64[D]")
+   TypeError: dtype=datetime64[D] is not supported. Supported resolutions are 's', 'ms', 'us', and 'ns'
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.0.rst:146]: With the `okexcept` above, this can be removed since the full traceback should be shown

**Comment by jbrockmendel** [comment]
@jreback little help with the docbuild?  (also thoughts on the API change?)

**Review by jreback — COMMENTED** [review]
suggestion on doc build. api change lgtm.

**Comment by jbrockmendel** [comment]
@jreback still struggling with the docbuild

File: doc/source/whatsnew/v2.0.0.rst (hunk 1 above)
Comment (mroeschke) [reply doc/source/whatsnew/v2.0.0.rst:134]: This section should just be

```
*New behavior*:

.. ipython:: python

    pd.Series(["2016-01-01"], dtype="datetime64[s]")
```

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst]: That did it, thanks

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Comment by jbrockmendel** [comment]
sweet!  the only big constructor-behavior-matching one left is #49348

**Pull request merged**